src/main/java/org/verapdf/wcag/algorithms/semanticalgorithms/utils/NodeUtils.java

```diff
@@ -99,6 +102,22 @@ public static double headingProbability(SemanticTextNode textNode, INode neighbo
 		} else if (!isUpperCaseString(textNode.getValue()) && isUpperCaseString(neighborTextNode.getValue())) {
 			probability -= HEADING_PROBABILITY_PARAMS[6];
 		}
+
+		if (StaticContainers.isDataLoader()) {
+			if (isFarFromNeighbor(textNode, neighborTextNode, textNode.getFontSize())) {
+				probability += HEADING_PROBABILITY_FAR_FROM_NEIGHBOR_BOOST;
+			} else if (isFarFromNeighbor(textNode, neighborTextNode, textNode.getFontSize() / 2)) {
+				probability += HEADING_PROBABILITY_FAR_FROM_NEIGHBOR_BOOST / 2;
+			} else {
+				probability -= HEADING_PROBABILITY_FAR_FROM_NEIGHBOR_BOOST / 2;
+			}
+			if (isSingleLine(textNode) && isShorterThanNeighbor(textNode, neighborTextNode)) {
+				probability += HEADING_PROBABILITY_SINGLE_LINE_BOOST;
+			} else {
+				probability -= HEADING_PROBABILITY_SINGLE_LINE_BOOST;
```

Comment:
I'm not sure, that we should reduce probability here. We already use `getLinesNumberHeadingProbability`